test/UnbrandedProjects/Platform-OpenAI-TypeSpec/src/Generated/Internal/MultipartFormDataBinaryContent.cs

```diff
@@ -0,0 +1,197 @@
+            Add(new StringContent(value), name, filename, contentType);
+        }
+
+        public void Add(decimal content, string name, string filename = null, string contentType = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            string value = content.ToString("G", CultureInfo.InvariantCulture);
+            Add(new StringContent(value), name, filename, contentType);
+        }
+
+        public void Add(bool content, string name, string filename = null, string contentType = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            string value = content ? "true" : "false";
+            Add(new StringContent(value), name, filename, contentType);
+        }
+
+        public void Add(Stream content, string name, string filename = null, string contentType = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Add(new StreamContent(content), name, filename, contentType);
+        }
+
+        public void Add(byte[] content, string name, string filename = null, string contentType = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Add(new ByteArrayContent(content), name, filename, contentType);
+        }
+
+        public void Add(BinaryData content, string name, string filename = null, string contentType = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Add(new ByteArrayContent(content.ToArray()), name, filename, contentType);
+        }
+
+        private void Add(HttpContent content, string name, string filename, string contentType)
+        {
+            if (filename != null)
+            {
+                Argument.AssertNotNullOrEmpty(filename, nameof(filename));
+                AddFilenameHeader(content, name, filename);
+            }
+            if (contentType != null)
+            {
+                Argument.AssertNotNullOrEmpty(contentType, nameof(contentType));
+                AddContentTypeHeader(content, contentType);
+            }
+            _multipartContent.Add(content, name);
+        }
+
+        public static void AddFilenameHeader(HttpContent content, string name, string filename)
+        {
+            ContentDispositionHeaderValue header = new ContentDispositionHeaderValue("form-data") { Name = name, FileName = filename };
+            content.Headers.ContentDisposition = header;
+        }
+
+        public static void AddContentTypeHeader(HttpContent content, string contentType)
+        {
+            MediaTypeHeaderValue header = new MediaTypeHeaderValue(contentType);
```

Comment:
Why are we adding this in a separate method?  I don't believe this is needed and should be removed.

Reply:
Each part(not just file) of the multipart/form-data payload may set the `content-type` to indicate its content type(see following example). We need provide a way to set content-type of a part. 
![image](https://github.com/Azure/autorest.csharp/assets/10458143/71f587e5-25b9-4875-bd03-a96f5e908a0a)

Reply:
The BCL type will do this automatically for you.

Reply:
BCL type will give default one. The request is customer can specify the `content-type` to each part.